client.go

```diff
@@ -107,3 +108,36 @@ func (c *Client) ResetAllScenarios() error {
 
 	return nil
 }
+
+// FindFor finds requests that were made for a StubRule
+func (c *Client) FindRequestsFor(stubRule *StubRule) (map[string]interface{}, error) {
```

Comment:
for the sake of keeping this PR small returning a `map[string]interface{}` else we can work on modelling out the expected response as a struct

Reply:
It would be great if we had a structure, at least `requests []map[string]interface{}`

Reply:
ok will address this